package week_1;

class Logger {

    private static Logger instance;

    private Logger() {

    }

    public static Logger getInstance() {

        if (instance == null) {

            instance = new Logger();

        }

        return instance;

    }

    public void log(String message) {

        System.out.println("Log Message: " + message);

    }

}


public class Singleton_Pattern {

    public static void main(String[] args) {

        Logger logger1 = Logger.getInstance();

        Logger logger2 = Logger.getInstance();

        logger1.log("Application Started");

        logger2.log("Application Running");

        if (logger1 == logger2) {

            System.out.println("Only one Logger instance is created");

        }

        else {

            System.out.println("Different instances created");

        }

    }

}